src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,184 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var surfaceAtts = require('../surface/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the X coordinates of the vertices on X axis.'
+        ].join(' ')
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Y coordinates of the vertices on Y axis.'
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Z coordinates of the vertices on Z axis.'
+        ].join(' ')
+    },
+    volume: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the 4th dimension of the vertices.'
+        ].join(' ')
+    },
+    isocap: {
+        valType: 'boolean',
+        role: 'info',
+        editType: 'calc',
+        dflt: false,
+        description: [
+            'Enable caps between minimum and maximum iso values.'
+        ].join(' ')
+    },
+    isovalue: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        dflt: [0],
+        description: [
+            'Sets iso surface boundaries.'
+        ].join(' ')
+    },
+    meshalgo: {
+        valType: 'string',
+        role: 'info',
+        dflt: 'MarchingCubes',
+        editType: 'calc',
+        description: [
+            'Sets the isosurface polygonizer algorithm:',
+            'including: `MarchingCubes` (i.e. default),',
+            '`MarchingTetrahedra` or `SurfaceNets`.'
+        ].join(' ')
+    },
+
+    text: {
+        valType: 'string',
+        role: 'info',
+        dflt: '',
+        arrayOk: true,
+        editType: 'calc',
+        description: [
+            'Sets the text elements associated with the vertices.',
+            'If trace `hoverinfo` contains a *text* flag and *hovertext* is not set,',
+            'these elements will be seen in the hover labels.'
+        ].join(' ')
+    },
+
+    // Color field
+    color: {
```

Comment:
Do we really need `color`, `vertexcolor` and `facecolor`? Should applying a colorscale to `value` (your `volume`) be enough?

Reply:
Reduced color attributes in https://github.com/plotly/plotly.js/pull/3340/commits/49aa7468d0ea7917e1917c49620c964332b1e0c2